skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -0,0 +1,228 @@
+import butterknife.ButterKnife;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import timber.log.Timber;
+
+/**
+ * Receive activity, for testing, needs refactor.
+ *
+ * @author huangyz0918 (huangyz0918@gmail.com)
+ */
+public class BtReceiverActivity extends InjectableActivity implements
+        BluetoothReceiver.BluetoothReceiverListener, BluetoothListAdapter.OnDeviceClickListener {
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.btn_refresh)
+    Button btnRefresh;
+
+    @BindView(R.id.list_bt_device)
+    RecyclerView recyclerView;
+
+    @Inject
+    ReceiverService receiverService;
+
+    @Inject
+    RxEventBus rxEventBus;
+
+    @Inject
+    BaseSchedulerProvider schedulerProvider;
+
+    private BluetoothAdapter bluetoothAdapter;
+    private BluetoothReceiver bluetoothReceiver;
+    private final BluetoothListAdapter bluetoothListAdapter = new BluetoothListAdapter(this);
+    private boolean isConnected = false;
+    private ProgressDialog progressDialog;
+
+    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_bt_receive);
+        ButterKnife.bind(this);
+        initEvents();
+
+        setTitle(getString(R.string.connect_bluetooth_title));
+        setSupportActionBar(toolbar);
+
+        // checking for if bluetooth enabled
+        if (!BluetoothUtils.isBluetoothEnabled()) {
+            BluetoothUtils.enableBluetooth();
+            updateDeviceList();
+        }
+    }
+
+    /**
+     * Add the bounded bluetooth devices.
+     */
+    public void addPairedDevices() {
+        Set<BluetoothDevice> bondedDevices = bluetoothAdapter.getBondedDevices();
+        if (bondedDevices != null) {
+            bluetoothListAdapter.addDevices(bondedDevices);
+        }
+    }
+
+    /**
+     * Rescan the bluetooth devices and update the list.
+     */
+    public void updateDeviceList() {
+        addPairedDevices();
+        if (!bluetoothAdapter.isDiscovering()) {
+            bluetoothAdapter.startDiscovery();
+        }
+        bluetoothListAdapter.notifyDataSetChanged();
+    }
+
+    private void initEvents() {
+        bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(recyclerView.getContext(),
+                linearLayoutManager.getOrientation());
+        recyclerView.addItemDecoration(dividerItemDecoration);
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.setAdapter(bluetoothListAdapter);
+        bluetoothReceiver = new BluetoothReceiver(this, this);
+        addPairedDevices();
+        BluetoothAdapter.getDefaultAdapter().startDiscovery();
```

Comment:
Either use `BluetoothAdapter.getDefaultAdapter()` everywhere or if you're assigning it to `bluetoothAdapter` then use the same variable everywhere it needs to be consistent everywhere.